src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (String(a) < String(b)) {
+        return -1;
+      } else if (String(a) > String(b)) {
+        return 1;
+      }
+
+      return 0;
+    },
+  ) {
+    for (let i = 0; i < this.length - 1; ) {
+      const a = this[i];
+      const b = this[i + 1];
+
+      if (compareFunction(a, b) > 0) {
+        this[i] = b;
+        this[i + 1] = a;
+        i = 0;
```

Comment:
It's good that you've implemented a swap when the compareFunction indicates the order is incorrect. However, resetting 'i' to 0 every time a swap occurs is inefficient. Consider using a different approach to avoid unnecessary iterations.